### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by rogpeppe**
We fix some of the more obvious errors, but there's a bunch of
work still remaining to be done.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/29/

File: docs/configuration.md [hunk 1]
```diff
@@ -1,77 +1,61 @@
...
-credentials makes the client all-powerful and as such these should be
-used with care. The use of this mechanism will eventually be phased
-out.
+(Required) This is the externally addressable location of the Candid server API.
+Candid needs to know its own address so that it can add third-party
+caveats addressed to itself and to create response addresses for identity
+providers such as OpenID that use browser redirection for communication.
+
+### mongo-addr, postgres-connection-string, ephemeral-storage
+
```
Comment (mhilton) [inline docs/configuration.md]: Please be consistent with the blank lines after headers (I think without)

File: docs/configuration.md [hunk 2]
```diff
@@ -95,6 +79,14 @@ identity provider in a given identity manager, in most case this does
 not make sense as the identity manager will only use the first one
 that is found.
 
+### Agent
+
```
Comment (mhilton) [inline docs/configuration.md]: see above

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/37/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

**Pull request merged**